docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md

```diff
@@ -0,0 +1,221 @@
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin/user, I would like a declarative and secure means by which to assign roles to my nodes by acting on Cluster.Topology metadata
+(for Clusters with ClusterClass).
+
+As a cluster admin/user, I would like a declarative and secure means by which to assign roles to my nodes by acting on KubeadmControlPlane
+MachineDeployments (for Clusters without ClusterClass).
+
+#### Story 2
+
+As a cluster admin/user, I would like to change labels or annotations on Machines without triggering Machine rollout.
+
+#### Story 3
+
+As a cluster admin/user, I would like to change nodeDrainTimeout on Machines without triggering Machine rollout.
+
+#### Story 4
+
+As a cluster admin/user, I would like to set autoscaler labels for MachineDeployments by changing Cluster.Topology metadata
+(for Clusters with ClusterClass).
+
+### Implementation Details/Notes/Constraints
+
+### Metadata propagation
+
+The following schema represent how metadata propagation works today (also documented in [book](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/metadata-propagation.html)).
+
+![Figure 1](./images/in-place-propagation/current-state.png)
+
+With this proposal we are suggesting to improve metadata propagation according as described in the following schema:
+
+![Figure 2](./images/in-place-propagation/proposed-changes.png)
+
+Following paragraphs provide more details about the proposed changes.
+
+#### Propagation of metadata from ClusterClass and Cluster Topology
+
+This change ensure that metadata from ClusterClass and Cluster Topology propagate consistently, in the sense that each set of labels and
+annotations will be applied in the same way to all the corresponding targets, more specifically:
+
+1. Metadata from ClusterClass.Spec.ControlPlane and Cluster.Spec.Topology.ControlPlane apply consistently to:
+- The ControlPlane object, e.g. KubeadmControlPlane (both top level metadata and spec.machineTemplate.metadata).
+- The InfrastructureMachineTemplate object eventually referenced by the ControlPlane (top level metadata only).
+
+2. Metadata from ClusterClass.Spec.Workers.MachineDeployments and Cluster.Spec.Workers.Topology.ControlPlane apply consistently to:
+- The MachineDeployment objects (both top level metadata and spec.template.metadata).
+- The InfrastructureMachineTemplate object referenced by the MachineDeployment (top level metadata only).
+- The BottstrapTemplate object referenced by the MachineDeployment (top level metadata only).
+
+Please note that above changes solves [Labels and annotations for MachineDeployments and KubeadmControlPlane created by topology controller](https://github.com/kubernetes-sigs/cluster-api/issues/7006);
+also, we are stopping to propagate existing labels on ClusterClass templates to objects created from them in favor of always
+propagating the same set of labels as described above.
```

Comment:
That's a great catch, I have missed because it was not documented in https://cluster-api.sigs.k8s.io/developer/architecture/controllers/metadata-propagation.html; updating diagram accordingly

Reply:
I have updated diagrams with metadata propagation from templates, dropping this point while we identify next steps on the improved picture